src/network/edit-settings.ts

```diff
@@ -171,78 +172,16 @@ export const editSettings = (
 		useAccountStore.setState((s: AccountState) => ({
 			settings: {
 				...s.settings,
-				prefs: reduce(
-					mods.prefs,
-					(acc, pref, key) => ({
-						...acc,
-						[key]: pref as string
-					}),
-					s.settings.prefs
-				),
-				props: reduce(
-					mods.props,
-					(acc, { app, value }, key) => {
-						const propIndex = findIndex(acc, (p) => p.name === key && p.zimlet === app);
-						if (propIndex >= 0) {
-							// eslint-disable-next-line no-param-reassign
-							acc[propIndex] = {
-								name: key,
-								zimlet: app,
-								_content: value as string
-							};
-						} else {
-							acc.push({
-								name: key,
-								zimlet: app,
-								_content: value as string
-							});
-						}
-						return acc;
-					},
-					s.settings.props
-				)
+				prefs: mergePrefs(mods, s),
+				props: mergeProps(mods, s)
 			},
 			account: {
 				...s.account,
 				displayName:
 					find(mods?.identity?.modifyList, (item) => item.id === s?.account?.id)?.prefs
 						.zimbraPrefIdentityName || s.account?.displayName,
 				identities: {
-					identity:
-						typeof s.account !== 'undefined'
-							? reduce(
-									mods?.identity?.modifyList,
-									(acc, { id, prefs }) => {
-										const propIndex = findIndex(
-											acc,
-											(itemMods, indexAccount) => acc[indexAccount].id === id
-										);
-										if (propIndex > -1) {
-											// eslint-disable-next-line no-param-reassign
-											acc[propIndex]._attrs = {
-												...acc[propIndex]._attrs,
-												...prefs
-											};
-											if (prefs.zimbraPrefIdentityName && acc[propIndex].name !== 'DEFAULT') {
-												// eslint-disable-next-line no-param-reassign
-												acc[propIndex].name = prefs.zimbraPrefIdentityName;
-											}
-										}
-										return acc;
-									},
-									[
-										...filter(
-											s.account.identities.identity,
-											(item) => !mods?.identity?.deleteList?.includes(item.id)
-										).filter((i) => i.name !== 'DEFAULT'),
-										...map(r?.CreateIdentityResponse, (item) => item.identity[0]),
-										...filter(
-											s.account.identities.identity,
-											(item) => !mods?.identity?.deleteList?.includes(item.id)
-										).filter((i) => i.name === 'DEFAULT')
-									]
-								)
-							: undefined
+					identity: updateIdentities(s, mods, r)
 				}
 			} as Account
```

Comment:
We could replace the entire block with the updateSettings and updateAccount functions, couldn't we?

Reply:
We changed this by mistake, because we didn't notice it was deprecated code. Thanks for spotting that out. For this reason we reverted the change. Is it ok for you?